bin/generate_truss_examples.py

```diff
@@ -0,0 +1,301 @@
+"""
+Script to take the Truss examples in https://github.com/basetenlabs/truss-examples-2,
+and generate documentation.
+
+
+
+Usage:
+```
+$ poetry run python bin/generate_truss_examples.py
+```
+"""
+import enum
+import json
+import os
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import yaml
+
+DOC_CONFIGURATION_FILE = "doc.yaml"
+TRUSS_EXAMPLES_REPO = "https://github.com/basetenlabs/truss-examples"
+DESTINATION_DIR = "truss-examples"
+MINT_CONFIG_PATH = "docs/mint.json"
+
+
+class FileType(enum.Enum):
+    YAML = "yaml"
+    PYTHON = "python"
+
+
+def clone_repo():
+    """
+    If the destination directory exists, remove it.
+    Then, clone the given repo into the specified directory.
+    """
+    if os.path.exists(DESTINATION_DIR):
```

Comment:
nit: prefer to use `pathlib.Path` operations over `os.path` to stay consistent with the style of the rest of the repo